### PR TITLE
AMP SSR dependencies

### PR DESCRIPTION
**Pull request by dijs**
- [x] Implement Headers class with added Proxy'ing of header keys using getter (with deprecation warning)
- [x] Use the Headers class from the xdn module
- [x] Move fetch into xdn
- [x] Alias cross-fetch to xdn/fetch in server
- [x] RSF should use cross-fetch
- [x] Take care of webpack warning
- [x] Use an internal Headers class and wrap with proxy for `direct` access to values
- [x] Remove `cross-fetch`
- [x] Try to make warning suppression more specific

### CODE REVIEW

File: packages/react-storefront/src/fetch.js [hunk 1]
```diff
@@ -184,10 +184,11 @@ export default function fetch(url, options = {}, qsOptions) {
             status: response.statusCode,
             statusText: response.statusText,
             ok,
-            headers: response.headers,
+            headers: { ...response.headers, get: key => response.headers[key] },
```
Comment (dijs) [inline packages/react-storefront/src/fetch.js]: Implement Headers class with added Proxy'ing of header keys using getter - with deprecation warning.

Comment (dijs) [reply packages/react-storefront/src/fetch.js]: Use the Headers class from the xdn module

Comment (dijs) [reply packages/react-storefront/src/fetch.js]: - Move fetch into xdn
- Alias cross-fetch to xdn/fetch
- RSF should import fetch from cross-fetch
- Just makes sure express boilerplate still works

File: packages/react-storefront/webpack/common.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ function createClientConfig(
     ),
     resolve: {
       alias: Object.assign({}, createAliases(root), alias, {
-        fetch: 'isomorphic-unfetch'
+        fetch: 'cross-fetch'
```
Comment (markbrocato) [inline packages/react-storefront/webpack/common.js]:
```suggestion
        fetch: 'isomorphic-unfetch'
```

File: packages/react-storefront/package.json [hunk 3]
```diff
@@ -87,6 +87,7 @@
     "clean-webpack-plugin": "^1.0.0",
     "cookie": "^0.3.1",
     "copy-webpack-plugin": "^4.5.1",
+    "cross-fetch": "^3.0.4",
```
Comment (markbrocato) [inline packages/react-storefront/package.json]: this can be removed

File: packages/react-storefront-moov-xdn/test/fetch.common.test.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
-import fetch from '../src/fetch'
+// import fetch from '../src/fetch'
```
Comment (markbrocato) [inline packages/react-storefront-moov-xdn/test/fetch.common.test.js]: this can be removed

**Comment by markbrocato** [comment]
Good, just needs a changelog:

"The headers object included in fetch responses now matches the [Headers](https://developer.mozilla.org/en-US/docs/Web/API/Headers) interface from the standard fetch API."

**Pull request merged**